src/ScreenshotWindow.vala

```diff
@@ -321,27 +321,25 @@ namespace Screenshot {
 
             if (from_command == false) {
                 var save_dialog = new Screenshot.Widgets.SaveDialog (screenshot, settings, this);
-                save_dialog.save_response.connect ((response, folder_dir, output_name, format) => {
+                save_dialog.save_response.connect ((folder_dir, output_name, format) => {
                     save_dialog.destroy ();
```

Comment:
I am a little uncomfortable with the save_dialog being destroyed in two different places, depending on the response.  It seems cleaner to me for it to be always destroyed by the function that created it (which would mean returning the response as before, or e.g returning null folder_dir as an indication to cancel save).

You could use the native dialog response signal and retreive folder_dir, outputname, format as properties of SaveDialog before destroying it.

You then separate the business logic from the UI logic.